sql/table.go

```diff
@@ -175,6 +175,76 @@ func (p *planner) getTableDesc(qname *parser.QualifiedName) (*TableDescriptor, e
 	return &desc, nil
 }
 
+// getTableLease acquires a lease for the specified table. The lease will be
+// released when the planner closes.
+func (p *planner) getTableLease(qname *parser.QualifiedName) (*TableDescriptor, error) {
+	if err := qname.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+
+	if qname.Database() == SystemDB.Name {
+		// We don't go through the normal lease mechanism for system tables. The
+		// system.lease table, in particular, is problematic because it is used for
+		// acquiring leases itself, creating a chicken&egg problem.
+		if tableDesc, ok := systemTables[qname.Table()]; ok {
+			return tableDesc, nil
+		}
+	}
+
+	tableID, err := p.getTableID(qname)
+	if err != nil {
+		return nil, err
+	}
+
+	if p.leases == nil {
+		p.leases = make(map[ID]*LeaseState)
+	}
+	if lease, ok := p.leases[tableID]; ok {
```

Comment:
I've made a minor reorg to this code which makes changing this to an `else if` more awkward. Avoiding the unnecessary map lookup is a premature optimization.